sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/LocalCryptographyClient.cs

```diff
@@ -62,44 +62,84 @@ protected LocalCryptographyClient()
+        /// <exception cref="ArgumentException">The specified <paramref name="algorithm"/> does not match the key corresponding to the key identifier.</exception>
+        /// <exception cref="CryptographicException">The local cryptographic provider threw an exception.</exception>
+        /// <exception cref="InvalidOperationException">The key is invalid for the current operation.</exception>
+        /// <exception cref="NotSupportedException">The operation is not supported with the specified key.</exception>
+        public virtual EncryptResult Encrypt(EncryptionAlgorithm algorithm, byte[] plaintext, CancellationToken cancellationToken = default) =>
+            Encrypt(algorithm, new EncryptOptions(plaintext), cancellationToken);
+
+        /// <summary>
+        /// Encrypts plaintext.
+        /// </summary>
+        /// <param name="algorithm">The <see cref="EncryptionAlgorithm"/> to use.</param>
+        /// <param name="options">An <see cref="EncryptOptions"/> containing the data to encrypt and other options for <paramref name="algorithm"/>-dependent encryption.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> to cancel the operation.</param>
+        /// <returns>
+        /// The result of the encrypt operation. The returned <see cref="EncryptResult"/> contains the encrypted data
+        /// along with all other information needed to decrypt it. This information should be stored with the encrypted data.
+        /// </returns>
+        /// <exception cref="ArgumentException">The specified <paramref name="algorithm"/> does not match the key corresponding to the key identifier.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="options"/> is null.</exception>
+        /// <exception cref="CryptographicException">The local cryptographic provider threw an exception.</exception>
+        /// <exception cref="InvalidOperationException">The key is invalid for the current operation.</exception>
+        /// <exception cref="NotSupportedException">The operation is not supported with the specified key.</exception>
+        public virtual async Task<EncryptResult> EncryptAsync(EncryptionAlgorithm algorithm, EncryptOptions options, CancellationToken cancellationToken = default)
         {
+            Argument.AssertNotNull(options, nameof(options));
+
             EncryptResult result = null;
             if (_provider.SupportsOperation(KeyOperation.Encrypt))
             {
-                result = await _provider.EncryptAsync(algorithm, plaintext, cancellationToken).ConfigureAwait(false);
+                result = await _provider.EncryptAsync(algorithm, options, cancellationToken).ConfigureAwait(false);
             }
 
             return result ?? throw LocalCryptographyProvider.CreateOperationNotSupported(nameof(KeyOperation.Encrypt));
         }
 
         /// <summary>
-        /// Encrypts the specified plain text.
+        /// Encrypts plaintext.
```

Comment:
nite: inconsistent with "Encrypts the specified plain text."

Reply:
See above.